### ISSUE TITLE
no thumbnail / resized image are generated in /assets/images

### ISSUE THREAD

**Issue by volkerrichert** [contao/contao#1090]
**Affected version(s)**
4.8.5, no other tested yet

**Description**
I try to upload some new images in a fresh 4.8.5 installation. Uploading via D&D onto the folder in the tree works fine, but no Thumb ist displayed just the "jpg" icon.

Clicking the pen next to ne new file will raise an error linke "Warning: file_get_contents(/var/www/html/assets/images/1/Standortanalysen_1920x200-303214b3.jpg): failed to open stream: No such file or directory".

Displaying the resized images in frontend are blockd by a 404. So i chekced the assets/image folder. The directories (0....) are created, also the json inside "deferred" but no(!) resized image.

As fare as i can see, all this images should be creted by Contao\CoreBundle\Image\ImageFactory::create and the third parameter $option is used to set a targetpath | ResizeOptions.  But nearly non call is using the third parameter, i found just one beside some test. But without this the Contao\Image\Resizer will not create any file.

```php
        if (null !== $options->getTargetPath()) {
            $this->filesystem->copy($image->getPath(), $options->getTargetPath(), true);
            $image = $this->createImage($image, $options->getTargetPath());
        }
```

**How to reproduce**
Just upload a file and check /assets/image

**Comment by asaage** [comment]
check this thread: https://community.contao.org/de/showthread.php?76618-Nach-Update-von-4-7-x-auf-aktuelle-4-8-4-Bildercache-quot-defekt-quot-gt-assets-images
If you are using the original 4.8.5 .htaccess this problem is most likely caused by a reverse-proxy that tries to serve static resources (pictures) directly.
That seems to be a default plesk webhosting configuration and does not work with the deferred-image-resize lately introduced in contao.

**Comment by volkerrichert** [comment]
i'm using the default .htaccess file and the ImageController is called and ist running on my local system ATM inside a docker container

**Comment by volkerrichert** [comment]
ok, i found the reason. my GDlib doesn't support jpeg, png works as expected. 

But contao should have a fallback handling if deferred image handling is failing, e.g. using the real image

**Comment by leofeyer** [comment]
@ausi WDYT?

**Comment by ausi** [comment]
I don’t think using the original image would be a good solution here.

How about a compiler pass that verifies that all image types from `contao.image.valid_extensions` can be handled by the Imagine service?

Sadly the Imagine interface doesn’t seem to have a method for that so we would need to use a test image for every format.

**Comment by Toflar** [comment]
You would also need a real image of the respective types to test for that, wouldn't you? I mean, just because you give a zero byte image the `.jpeg` extension doesn't mean, the lib can actually handle it.

I had the same issue and debugging it is not that easy. Maybe we can log the actually issue and have a general placeholder image saying "check your log files" (maybe with svg `<text>` or so?)?

**Comment by aschempp** [comment]
> Maybe we can log the actually issue and have a general placeholder image saying "check your log files" (maybe with svg <text> or so?)?

Not sure I like that this would show up in production …

**Comment by Toflar** [comment]
So you prefer having 500 server errors for image requests?

**Comment by ausi** [comment]
> So you prefer having 500 server errors for image requests?

I do. It is a server error, so it should be reported as such.

> You would also need a real image of the respective types to test for that

Yes. I’d use the image files from https://github.com/mathiasbynens/small for this.

**Comment by leofeyer** [comment]
> > So you prefer having 500 server errors for image requests?
> 
> I do. It is a server error, so it should be reported as such.

ACK.

**Comment by volkerrichert** [comment]
500 is fine.

**Comment by ausi** [comment]
What should be the solution now?

1. Compiler pass that verifies `contao.image.valid_extensions` and generates warnings for unsupported image types
2. Generate 500-Error responses as soon as an an image is tried to get resized? (would be in most cases the image request itself)

I think we should do both.

**Comment by leofeyer** [comment]
I think so, too. Will you make a PR?

**Comment by ausi** [comment]
Yes.

**Comment by ausi** [comment]
Closed in favor of #1814 and #1815
